data/blog/gn.md

```diff
@@ -0,0 +1,44 @@
+---
+title: "Using GN to Build Electron"
+author: nornagon
+date: '2018-09-05'
+---
+
+Electron now uses GN to build itself. Here's a discussion of why.
+
+---
+
+# GYP and GN
+
+When Electron was first released in 2013, Chromium's build configuration was written with [GYP], short for "Generate Your Projects".
+
+In 2014, the Chromium project introduced a new build configuration tool called [GN] (short for "Generate [Ninja]") Chromium's build files were migrated to GN and GYP was removed from the source code.
+
+Electron has historically kept a separation between the main [Electron code] and [libchromiumcontent], the part of Electron that wraps Chromium's 'content' submodule. Electron has carried on using GYP, while libchromiumcontent -- as a subset of Chromium -- switched to GN when Chromium did.
+
+Like gears that don't quite mesh, there was friction between using the two build systems. Maintaining compatibility was error-prone, from compiler flags and `#defines` that needed to be meticulously kept in sync between Chromium, Node, V8, and Electron.
+
+To address this, the Electron team has been working on moving everything to GN. Today, the [commit](https://github.com/electron/electron/pull/14097) to remove the last of the GYP code from Electron was landed in master.
+
+# What this means for you
+
+If you're contributing to Electron itself, the process of checking out and building Electron is very different. See [GN build instructions] for details.
```

Comment:
We should add something here or elsewhere in the post to mention that this only applies to master and not to older branches (3-0-x and older).